app/rdf/query-termsets.ts

```diff
@@ -0,0 +1,82 @@
+import groupBy from "lodash/groupBy";
+import ParsingClient from "sparql-http-client/ParsingClient";
+
+import { ComponentTermsets, Termset } from "@/domain/data";
+import { buildLocalizedSubQuery } from "@/rdf/query-utils";
+
+export const queryAllTermsets = async (options: {
+  locale: string;
+  sparqlClient: ParsingClient;
+}): Promise<{ termset: Termset; count: number }[]> => {
+  const { sparqlClient, locale } = options;
+  const qs = await sparqlClient.query.select(
+    `PREFIX cube: <https://cube.link/>
+PREFIX meta: <https://cube.link/meta/>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX schema: <http://schema.org/>
+PREFIX sh: <http://www.w3.org/ns/shacl#>
+
+SELECT DISTINCT (COUNT(distinct ?iri) as ?count) ?termsetIri ?termsetLabel WHERE {
+    ?iri cube:observationConstraint/sh:property ?dimension .
+    ?dimension a cube:KeyDimension .
+    ?dimension sh:in/rdf:rest*/rdf:first ?value.
+    ?value schema:inDefinedTermSet ?termsetIri .
+    ?termsetIri a meta:SharedDimension .
+    ${buildLocalizedSubQuery("termsetIri", "schema:name", "termsetLabel", { locale })}
+  }      GROUP BY ?termsetIri ?termsetLabel`
+  );
+
+  return qs.map((result) => ({
+    count: Number(result.count.value),
+    termset: {
+      iri: result.termsetIri.value,
+      label: result.termsetLabel.value,
+    },
+  }));
+};
+
+export const getCubeComponentTermsets = async (
+  iri: string,
+  options: {
```

Comment:
Not 100% sure, but looks like it's not used anywhere? Is it for debugging purposes? 👀

Reply:
It was there because I copied paste a previous file to start this file :facepalm:.